v-next/hardhat/test/internal/cli/telemetry/sentry/anonymizer.ts

```diff
@@ -17,6 +19,29 @@ class MockedAnonymizer extends Anonymizer {
 }
 
 describe("Anonymizer", () => {
+  it("should clone key information from an anonymized event", async () => {
+    const originalEvent: Event = {
+      event_id: "my-event",
+      platform: "platform1",
+      release: "release1",
+      timestamp: 1754398906,
+      extra: {
+        another: "example",
+      },
+    };
+
+    const anonymizer = new Anonymizer();
+
+    const result = await anonymizer.anonymize(originalEvent);
+
+    if (!result.success) {
+      assert.fail("The event should anonymize without issue");
+      return;
+    }
+
+    assert.deepEqual(result.event, originalEvent);
```

Comment:
Any suggestions of a better pattern here would be welcome. Or would we use an invariant?